libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestResponse.java

```diff
@@ -170,7 +170,10 @@ public String getContentType() {
 	public String asString() throws IOException {
 		if (responseAsString == null) {
 			byte[] bytes = asBytes(); // will also compute responseCharSet
-			responseAsString = new String(bytes, responseCharSet);
+			if(responseAsString==null){
```

Comment:
I'm not sure I understand what the intent of this is. The `if (responseAsString == null)` check is already being performed in `line 171`, and calling `asBytes()` doesn't affect this (`asBytes()` only returns `responseAsBytes`, it doesn't mutate the value of `responseAsString`). This check is redundant and doesn't perform anything as far as I can tell.

Reply:
```
JsonObject jsonObject = (JsonObject) jsonParser.parse(response.asString());

public String asString() throws IOException {
	if (responseAsString == null) {
		byte[] bytes = asBytes(); // will also compute responseCharSet
		///////////////////////////
		asBytes() 151 line call consume()()
		consume()() 119 line
		if (responseAsBytes != null && responseAsBytes.length > 0) {
			responseAsString = new String(responseAsBytes, responseCharSet);
		}
		///////////////////////////
		responseAsString = new String(bytes, responseCharSet); //double new String
	}
	return responseAsString;
}
```


I have a lot of large amount of data downloading

Reply:
For a large dataset, you probably want to use `Jackson` to stream parse the JSON response, instead of calling `asString()` and using `JSONObject`.